### PR TITLE
[MU4] sampler_api_updates

### PR DESCRIPTION
**Pull request by vpereverzev**

### CODE REVIEW

**Review by RomanPudashkin — APPROVED** [review]

File: src/framework/musesampler/internal/libhandler.h [hunk 1]
```diff
@@ -140,24 +195,53 @@ struct MuseSamplerLibHandler
...
+        addNoteEvent = (ms_MuseSampler_add_track_note_event)dlsym(m_lib, "ms_MuseSampler_add_track_note_event");
+
+        isRangedArticulation = (ms_MuseSampler_is_ranged_articulation)dlsym(m_lib, "ms_MuseSampler_is_ranged_articulation");
+        addTrackEventRangeStart = (ms_MuseSampler_add_track_event_range_start)dlsym(m_lib, "ms_MuseSampler_add_track_event_range_start");
+        addTrackEventRangeEnd = (ms_MuseSampler_add_track_event_range_end)dlsym(m_lib, "ms_MuseSampler_add_track_event_range_end");
 
+        startAuditionMode = (ms_MuseSampler_start_audition_mode)dlsym(m_lib, "ms_MuseSampler_start_audition_mode");
+        stopAuditionMode = (ms_MuseSampler_stop_audition_mode)dlsym(m_lib, "ms_MuseSampler_stop_audition_mode");
+        startAuditionNote = (ms_MuseSampler_start_audition_note)dlsym(m_lib, "ms_MuseSampler_start_audition_note");
+        stopAuditionNote = (ms_MuseSampler_stop_audition_note)dlsym(m_lib, "ms_MuseSampelr_stop_audition_note");
```
Comment (cbjeukendrup) [inline src/framework/musesampler/internal/libhandler.h:239]: Typo?

File: src/framework/musesampler/internal/libhandler.h [hunk 2]
```diff
@@ -140,24 +195,53 @@ struct MuseSamplerLibHandler
...
+
+        startLivePlayMode = (ms_MuseSampler_start_liveplay_mode)dlsym(m_lib, "ms_MuseSampler_start_liveplay_mode");
+        stopLivePlayMode = (ms_MuseSampler_stop_liveplay_mode)dlsym(m_lib, "ms_MuseSampler_stop_liveplay_mode");
+        startLivePlayNote = (ms_MuseSampler_start_liveplay_note)dlsym(m_lib, "ms_MuseSampler_start_liveplay_note");
+        stopLivePlayNote = (ms_MuseSampler_stop_liveplay_note)dlsym(m_lib, "ms_MuseSampelr_stop_liveplay_note");
```
Comment (cbjeukendrup) [inline src/framework/musesampler/internal/libhandler.h:244]: Typo?

File: src/framework/musesampler/internal/libhandler.h [hunk 3]
```diff
@@ -37,28 +37,26 @@ namespace mu::musesampler {
 struct MuseSamplerLibHandler
 {
     ms_Result initLib() { return ms_init(); }
+    bool containsInstrument(const char* mpe_id, const char* musicxml)
+    {
+        return ms_contains_instrument(mpe_id, musicxml) == 1;
+    }
+    int getMatchingInstrumentId(const char* pack, const char* name) { reutrn ms_get_matching_instrument_id(pack, name); }
```
Comment (cbjeukendrup) [inline src/framework/musesampler/internal/libhandler.h:45]: Typo? (`reutrn`)

File: src/framework/musesampler/internal/libhandler.h (hunk 1 above)
Comment (RobFog) [reply src/framework/musesampler/internal/libhandler.h:239]: "MuseSampelr" must be a typo.

File: src/framework/musesampler/internal/libhandler.h (hunk 2 above)
Comment (RobFog) [reply src/framework/musesampler/internal/libhandler.h:244]: Here also

**Pull request merged**